src/vs/platform/lifecycle/electron-main/lifecycleMainService.ts

```diff
@@ -550,9 +551,33 @@ export class LifecycleMainService extends Disposable implements ILifecycleMainSe
 		this.quit().then(veto => quitVetoed = veto);
 	}
 
-	kill(code?: number): void {
+	async kill(code?: number): Promise<void> {
 		this.logService.trace('Lifecycle#kill()');
 
+		// API tests: we have seen issues (native crashes) when calling app.exit()
+		// without closing the test window properly. app.exit() is not recommended
+		// when having a window open, so for tests running we make sure to close the
+		// window. Since we do not want to block the tests from completing longer
+		// than needed, we set a race of 1s to exit anyway.
+		//
+		// Note: Electron implements a similar logic here:
+		// https://github.com/electron/electron/blob/fe5318d753637c3903e23fc1ed1b263025887b6a/spec-main/window-helpers.ts#L5
+		//
+		if (this.environmentService.isExtensionDevelopment && !!this.environmentService.extensionTestsLocationURI && this.windowCounter === 1) {
+			await Promise.race([
+				timeout(1000),
+				new Promise(c => {
+					const testWindow = BrowserWindow.getAllWindows()[0];
+					if (testWindow && !testWindow.isDestroyed()) {
+						testWindow.on('closed', () => c());
```

Comment:
Would also want to check the validity of `testWindow.webContents` and listen to the event accordingly for the reasons mentioned here https://github.com/electron/electron/blob/master/spec-main/window-helpers.ts#L7-L20

Reply:
@deepak1556 ok see my latest commit